patches/elasticpress.0003.fix.php-warning-for-cli-batch-time-length-calculation.patch

```diff
@@ -1,26 +1,26 @@
-From efe35a4b7b545788876f3ec3d8281d5f7c47826e Mon Sep 17 00:00:00 2001
-From: Nabi <artsanaat@gmail.com>
-Date: Thu, 29 Feb 2024 14:21:18 +0330
-Subject: [PATCH] Fixed PHP warning for batch time length calculation in
- ElasticPress CLI.
+From b377a1313cb023c022167921038dbab8cc8aaee4 Mon Sep 17 00:00:00 2001
+From: navidabdi <artsanaat@gmail.com>
+Date: Fri, 21 Mar 2025 12:47:11 +0400
+Subject: [PATCH] fix: PHP warning for batch time length calculation in
+ ElasticPress CLI
 
 ---
  includes/classes/Command.php | 2 +-
  1 file changed, 1 insertion(+), 1 deletion(-)
 
 diff --git a/includes/classes/Command.php b/includes/classes/Command.php
-index 81fa2f2c5..a1c6c2e93 100644
+index 0ae7e9bcd..55b30596d 100644
 --- a/includes/classes/Command.php
 +++ b/includes/classes/Command.php
-@@ -1335,7 +1335,7 @@ class Command extends WP_CLI_Command {
+@@ -1331,7 +1331,7 @@ class Command extends WP_CLI_Command {
  		if ( 'index_next_batch' === $context ) {
- 			$counter++;
+ 			++$counter;
  			if ( ( $counter % 10 ) === 0 ) {
--				$time_elapsed_diff = $time_elapsed > 0 ? ' (+' . (string) ( Utility::timer_stop() - (int) $time_elapsed ) . ')' : '';
-+        $time_elapsed_diff = $time_elapsed > 0 ? ' (+' . (string) ( Utility::timer_stop() - (int) $time_elapsed ) . ')' : '';
+-				$time_elapsed_diff = $time_elapsed > 0 ? ' (+' . (string) ( Utility::timer_stop() - $time_elapsed ) . ')' : '';
++				$time_elapsed_diff = $time_elapsed > 0 ? ' (+' . (string) ( Utility::timer_stop() - (int) $time_elapsed ) . ')' : '';
```

Comment:
Why do we need to typecast this? The current version initialises the variable as an `int` already

https://github.com/10up/ElasticPress/blob/5.1.4/includes/classes/Command.php#L1306

Reply:
This part of the code did not change for 4 years, and 
<img width="1531" alt="image" src="https://github.com/user-attachments/assets/38680461-76c5-40b0-afe6-1191e5478a1d" />

The original patch was created 3 years ago
<img width="1557" alt="image" src="https://github.com/user-attachments/assets/ba407a14-6d6d-4712-b771-711df71f9055" />

So I don't know maybe the type is changing somewhere during the code execution?

Reply:
The `Utility::timer_stop()` returns a float but subtraction should work still. Can you give the exact warning which you receive?

Reply:
@fabianmarz , this patch [was initially created via Bogdan](https://github.com/makers99/wp-cli-shared-patches/commit/5803c066ec1e71181de0d78641c952beb6b4e2df), so I don't know what was the exact error, I just updated the patch to be applicable again, if you think the issue has been resolved, then we can remove this patch totally (but I did not find any sign of resolving the issue via the vendor)

Reply:
Can you probably test the functionality locally to see if this is still applicable? Even you don't see it was "fixed" by the vendor, I'm not sure why this code adjustment exists at all since from the code everything seems to look fine.

Reply:
I can't see any issues via the ElasticPress CLI, and the sync is working fine without this patch, so I think we can remove the patch completely